src/script/script.h

```diff
@@ -412,6 +412,9 @@ class CScript : public CScriptBase
     CScript(std::vector<unsigned char>::const_iterator pbegin, std::vector<unsigned char>::const_iterator pend) : CScriptBase(pbegin, pend) { }
     CScript(const unsigned char* pbegin, const unsigned char* pend) : CScriptBase(pbegin, pend) { }
 
+    // Define a copy constructor so that operator= is a copy instead of a move to avoid undefined behavior
```

Comment:
could explain how this would otherwise be undefined behavior and why this can't be fixed in `prevector.h`?

Reply:
I'm not sure why it would be undefined behavior,  but UBSan was triggering on it. I think prevector already has a copy constructor?

Reply:
In that case I'd prefer to leave the third commit for a separate pull request. Changing those parts of the code without understanding why for the sole reason to please a sanitizer seems worrisome

Reply:
I've dropped the commit.